test/integration/targets/aws_ses_identity_dkim/tasks/main.yml

```diff
@@ -0,0 +1,128 @@
+---
+# tasks file for aws_ses_identity_dkim test
+
+- name: set up aws connection info
+  set_fact:
+    aws_connection_info: &aws_connection_info
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token }}"
+      region: "{{ aws_region }}"
+  no_log: yes
+
+- block:
+
+    ## prepare SES idenitiy to run tests on
+
+    - name: Create SES identity for running DKIM module tests on (domain identity)
+      aws_ses_identity:
+        identity: "{{ dkim_identity_domain }}"
+        state: present
+        <<: *aws_connection_info
+      register: ses_domain_identity
+
+    - name: Create SES identity for running DKIM module tests on (email identity)
+      aws_ses_identity:
+        identity: "{{ dkim_identity_email }}"
+        state: present
+        <<: *aws_connection_info
+      register: ses_email_identity
+
+    ## DKIM enable test
+
+    - name: Enable DKIM verification on domain identity
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_domain }}"
+        state: enabled
+        <<: *aws_connection_info
+      register: domain_result
+
+    - name: assert successful result
+      assert:
+        that:
+          - domain_result.changed == True
+          - domain_result.dkim_enabled == True
```

Comment:
The key here doesn't match the module.

The module currently returns all the atttributes wrapped in a dkim_attributes property so this would have to be:

    domain_resullt.dkim_attributes.dkim_enabled

But I'd suggest changing the module so it directly returns the attributes as I think the way it looks in this test is easier to read. 

That would require replacing line 168 in the module with

    module.exit_json(changed=changed, **camel_dict_to_snake_dict(attributes))

and updating the examples and return value documentation at the top of the module.

Reply:
Decided to stay with the current logic and fix the tests instead. As I conciously decided to keep the AWS return values in it's own dict back then, just forgot about the structure when writing the test. ;)
It seems kind of a convention for (AWS) Ansible modules nowadays to seperate aws return values from the things returned by Ansible directly (kind of seperate namespacing) and I kind of like that approach as it allows to go one level deeper to filter out all Ansible stuff and still see all AWS stuff at a glance.